web/src/app/util/TelTextField.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect, useState } from 'react'
+import TextField from '@material-ui/core/TextField'
+import { Check, Close } from '@material-ui/icons'
+import _ from 'lodash-es'
+import InputAdornment from '@material-ui/core/InputAdornment'
+import { makeStyles } from '@material-ui/core'
+import { DEBOUNCE_DELAY } from '../config'
+
+import gql from 'graphql-tag'
+import { useQuery } from '@apollo/react-hooks'
+
+const isValidNumber = gql`
+  query($number: String!) {
+    phoneNumberInfo(number: $number) {
+      id
+      valid
+    }
+  }
+`
+
+const useStyles = makeStyles({
+  valid: {
+    fill: 'green',
+  },
+  invalid: {
+    fill: 'red',
+  },
+})
+
+export default function TelTextField({ ...props }): JSX.Element {
```

Comment:
Missing type info:

```suggestion
export default function TelTextField(props: TextFieldProps & {value: string}): JSX.Element {
```

`TextFieldProps` comes from material UI, the `value:string` bit is because we require a string type here whereas the `TextField` defines it as unknown.